AirCastingTests/API/DormantStreamAlertAPITest.swift

```diff
@@ -0,0 +1,62 @@
+// Created by Lunar on 01/08/2022.
+//
+
+import XCTest
+@testable import AirCasting
+
+class DormantStreamAlertAPITest: APIServiceTestCase {
+    lazy var sut = DormantStreamAlertAPI()
+
+    func test_whenSendsRequest_hasCorrectURL() throws {
+            var receivedRequest: URLRequest?
+            client.requestTaskStub = { request, completion in
+                receivedRequest = request
+                completion(.failure(URLError(.timedOut)), request)
+            }
+        
+            sut.sendNewSetting(value: true, completion: { _ in })
+
+            let request = try XCTUnwrap(receivedRequest)
+            XCTAssertEqual(request.url, URL(string: "http://aircasting.org/api/user/settings")!)
+        }
+
+    func test_whenSendsRequest_usesPOSTMethod() throws {
+        var receivedRequest: URLRequest?
+        client.requestTaskStub = { request, completion in
+            receivedRequest = request
+            completion(.failure(URLError(.timedOut)), request)
+        }
+        
+        sut.sendNewSetting(value: true, completion: { _ in })
+        let request = try XCTUnwrap(receivedRequest)
+        XCTAssertEqual(request.httpMethod, "POST")
+    }
+
+    func test_whenSendsRequestwithTrueValue_hasCorrectBody() throws {
+        var receivedRequest: URLRequest?
+        client.requestTaskStub = { request, completion in
+            receivedRequest = request
+            completion(.failure(URLError(.timedOut)), request)
+        }
+        
+        sut.sendNewSetting(value: true, completion: { _ in })
+        let body = try XCTUnwrap(receivedRequest?.httpBody)
+        let expectedBody = "{\"data\":{\"session_stopped_alert\":\"true\"}}"
+        
+        XCTAssertEqual(String(data: body, encoding: .utf8), expectedBody)
+    }
+    
+    func test_whenSendsRequestWithFalseValue_hasCorrectBody() throws {
+        var receivedRequest: URLRequest?
+        client.requestTaskStub = { request, completion in
+            receivedRequest = request
+            completion(.failure(URLError(.timedOut)), request)
+        }
+        
+        sut.sendNewSetting(value: false, completion: { _ in })
+        let body = try XCTUnwrap(receivedRequest?.httpBody)
+        let expectedBody = "{\"data\":{\"session_stopped_alert\":\"false\"}}"
+        
+        XCTAssertEqual(String(data: body, encoding: .utf8), expectedBody)
+    }
```

Comment:
I am not any expert in those Tests, so just an idea. Do we need to check body content based on the the value which we are sending? Can't we just assume that the value is `.any` and then, aim to test only `body` itself - then we would be able to stay with only one out of those two tests. What do you think?

Reply:
But the body itself changes, so I'm not sure how could we change this to use `.any`? Could you elaborate?

Reply:
I can see now that I took the wrong way of thinking yesterday (thinking that maybe we could test what was not changed inside the body, but no, I think it better not to go this way) - sorry for the confusion.